src/csharp/Microsoft.Spark.E2ETest/Hadoop/FileSystemTests.cs

```diff
@@ -52,5 +53,28 @@ public void TestDelete()
 
             Assert.False(Directory.Exists(path));
         }
+
+        [Fact]
+        public void TestExists()
+        {
+            using FileSystem fs = FileSystem.Get(_spark.SparkContext.HadoopConfiguration());
+
+            using var tempDirectory = new TemporaryDirectory();
+            string path = Path.Combine(tempDirectory.Path, "temp-table");
+
+            _spark
+                .Range(25)
+                .Coalesce(1)
+                .Write()
+                .Csv(path);
+
+            Assert.True(fs.Exists(path));
+
+            var dataFile = Directory.GetFiles(path, "*.csv").FirstOrDefault();
+
+            Assert.NotNull(dataFile);
+
+            Assert.True(fs.Exists(dataFile));
```

Comment:
Maybe we should add an assertion to test that `fs.Exists()` returns `false` on a non-existent path?

For example you could do something like this:

```
string path = Path.Combine(tempDirectory.Path, "temp-table");

Assert.False(fs.Exists(path));
_spark.Range(25).Write().Csv(path);
Assert.True(fs.Exists(path));
```

That way, you prove that the `Exists()` return value changes after writing to that path.

Reply:
Great catch. I have added the assertion.

Reply:
Great catch. I have added the check.